testtools/matchers/_dict.py

```diff
@@ -235,15 +237,15 @@ class KeysEqual(Matcher):
     def __init__(self, *expected):
         """Create a `KeysEqual` Matcher.
 
-        :param expected: The keys the dict is expected to have.  If a dict,
-            then we use the keys of that dict, if a collection, we assume it
-            is a collection of expected keys.
+        :param expected: The keys the matchee is expected to have. As a
+            special case, if a single argument is specified, and it is a
+            mapping, then we use its keys as the expected set.
         """
         super(KeysEqual, self).__init__()
-        try:
+        if len(expected) == 1 and isinstance(expected[0], Mapping):
```

Comment:
Eek, `Mapping` does not work as I expected it to, so I've removed that, and gone to a simpler approach.

BTW, sorry for the long delay. I must have missed the notification from GitHub and this slipped my mind. Thanks for the review.